ee-feature-pack/common/src/main/resources/modules/system/layers/base/com/h2database/h2/main/module.xml

```diff
@@ -37,9 +37,14 @@
         <module name="java.naming"/>
         <module name="java.scripting"/>
         <module name="java.sql"/>
-        <!--WFLY-14219 Remove deprecated <module name="javax.api"/> -->
-        <module name="javax.transaction.api"/>
-        <module name="javax.servlet.api" optional="true"/>
+        <!-- H2 uses Jakarta Servlet to allow apps (in our case deployments) to expose its console as a servlet.
+             WildFly isn't concerning itself with allowing this H2 feature to continue to work in both a
+             javax.* EE namespace and jakarta.* EE namespace environment, so this no longer works out of the box.
+             But, we add an optional dep on a non-existent module as a hook to allow users to get it to work.
+             Users who want this to work can provide a module with the name:slot javax.servlet.api:h2. That module could
+             be a simple alias to the javax.servlet.api module the server provides, if that works with this module's
+             H2 artifact. Or the user could provide a non-alias module that packages and exposes the servlet resources H2 needs. -->
+        <module name="javax.servlet.api:h2" optional="true"/>
```

Comment:
@jamezp I tweaked this a bit since your review. Instead of flat out removing the previous javax.servlet.api dep, I added this hook to allow users who want this to work to provide the dep themselves.

Reply:
I'm guessing you reverted it back as it looks the same as I remember. If so this still looks good to me :)

Reply:
Hmm, reverting that wasn't something I meant to do!

Reply:
I restored it.

Reply:
That makes more sense and it looks good to me.